ionoscloud/resource_server.go

```diff
@@ -603,73 +617,82 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
-	firewallRules, apiResponse, err := client.FirewallRulesApi.DatacentersServersNicsFirewallrulesGet(ctx, datacenterId,
-		*serverReq.Id, *firstNicItem.Id).Execute()
-	logApiRequestTime(apiResponse)
-
-	if err != nil {
-		diags := diag.FromErr(fmt.Errorf("an error occurred while fetching firewall rules: %w", err))
-		return diags
-	}
-
-	if firewallRules.Items != nil {
-		if len(*firewallRules.Items) > 0 {
-			if err := d.Set("firewallrule_id", *(*firewallRules.Items)[0].Id); err != nil {
-				diags := diag.FromErr(err)
-				return diags
+	if foundServer.Entities.Nics.Items != nil {
+		if len(*foundServer.Entities.Nics.Items) > 0 {
+			foundFirstNic := (*foundServer.Entities.Nics.Items)[0]
+			var orderedRuleIds []string
+			if foundFirstNic.Entities != nil && foundFirstNic.Entities.Firewallrules != nil && foundFirstNic.Entities.Firewallrules.Items != nil {
+				sentFirstNic := (*serverReq.Entities.Nics.Items)[0]
+
+				if sentFirstNic.Entities != nil && sentFirstNic.Entities.Firewallrules != nil && sentFirstNic.Entities.Firewallrules.Items != nil {
+					sentRules := *sentFirstNic.Entities.Firewallrules.Items
+					foundRules := *foundFirstNic.Entities.Firewallrules.Items
+					orderedRuleIds = extractOrderedFirewallIds(foundRules, sentRules)
+					if len(orderedRuleIds) > 0 {
+						if err := d.Set("firewallrule_id", orderedRuleIds[0]); err != nil {
+							diags := diag.FromErr(err)
+							return diags
+						}
+					}
+				}
+			}
+			if len(orderedRuleIds) > 0 {
+				if err := d.Set("firewallrule_ids", orderedRuleIds); err != nil {
+					diags := diag.FromErr(err)
+					return diags
+				}
 			}
-		}
-	}
 
-	if firstNicItem.Id != nil {
-		err := d.Set("primary_nic", *firstNicItem.Id)
-		if err != nil {
-			diags := diag.FromErr(fmt.Errorf("error while setting primary nic %s: %w", d.Id(), err))
-			return diags
-		}
-	}
+			if foundFirstNic.Id != nil {
+				err := d.Set("primary_nic", *foundFirstNic.Id)
+				if err != nil {
+					diags := diag.FromErr(fmt.Errorf("error while setting primary nic %s: %w", d.Id(), err))
+					return diags
+				}
+			}
+			firstNicProps := foundFirstNic.Properties
+			if firstNicProps != nil {
+				firstNicIps := firstNicProps.Ips
+				if firstNicIps != nil &&
+					len(*firstNicIps) > 0 {
+					log.Printf("[DEBUG] set primary_ip to %s", (*firstNicIps)[0])
+					if err := d.Set("primary_ip", (*firstNicIps)[0]); err != nil {
+						diags := diag.FromErr(fmt.Errorf("error while setting primary ip %s: %w", d.Id(), err))
+						return diags
+					}
+				}
 
-	firstNicProps := firstNicItem.Properties
-	if firstNicProps != nil {
-		firstNicIps := firstNicProps.Ips
-		if firstNicIps != nil &&
-			len(*firstNicIps) > 0 {
-			log.Printf("[DEBUG] set primary_ip to %s", (*firstNicIps)[0])
-			if err := d.Set("primary_ip", (*firstNicIps)[0]); err != nil {
-				diags := diag.FromErr(fmt.Errorf("error while setting primary ip %s: %w", d.Id(), err))
-				return diags
+				volumeItems := serverReq.Entities.Volumes.Items
```

Comment:
We should be using orderedRuleIds here to keep the order.

Reply:
also yes, we should move to serverReq when setting properties

Reply:
@cristiGuranIonos not sure what you mean by your answers. More than that, I see that my comment doesn't make sense in the first place, we can't set the info about volumes using `foundFirstNic` because the `nic` doesn't contain information about volumes, I will mark this as solved for now since my original comment didn't make sense.